src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
> and advocate for letting jQuery be exactly as iterable as arrays sharing its environment (and hence not to check for Symbol.iterator at all).

Almost agree. If there were an environment that implemented `Symbol`, but hadn't implemented `Symbol.iterator` yet, you'd end up with `jQuery.fn.undefined = undefined`. If you don't care about that, then it's all good and I agree. Also, I don't know of any runtimes that match that, but for a few months both Chrome and Firefox did (obviously not anymore).